src/test/java/org/folio/rest/impl/IdpTest.java

```diff
@@ -0,0 +1,255 @@
+package org.folio.rest.impl;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.startsWith;
+import static org.hamcrest.Matchers.is;
+
+import io.restassured.RestAssured;
+import io.restassured.http.Cookie;
+import io.restassured.http.Header;
+import io.restassured.response.ExtractableResponse;
+import io.restassured.response.Response;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.regex.Pattern;
+import org.folio.config.SamlConfigHolder;
+import org.folio.rest.RestVerticle;
+import org.folio.util.MockJson;
+import org.folio.util.StringUtil;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+/**
+ * Test against a real IDP: https://simplesamlphp.org/ running in a Docker container.
+ */
+@RunWith(VertxUnitRunner.class)
+public class IdpTest {
+  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(IdpTest.class);
+  private static final boolean DEBUG = false;
+  private static final ImageFromDockerfile simplesamlphp =
+      new ImageFromDockerfile().withFileFromPath(".", Path.of("src/test/resources/simplesamlphp/"));
+
+  private static final String TENANT = "diku";
+  private static final Header TENANT_HEADER = new Header("X-Okapi-Tenant", TENANT);
+  private static final Header TOKEN_HEADER = new Header("X-Okapi-Token", "mytoken");
+  private static final Header JSON_CONTENT_TYPE_HEADER = new Header("Content-Type", "application/json");
+  private static final String STRIPES_URL = "http://localhost:3000";
+
+  private static final int MODULE_PORT = 8081;
+  private static final int OKAPI_PORT = 9130;
```

Comment:
```suggestion
  private static final int OKAPI_PORT = 9230;
```

If we are using hard-coded ports, I would like it to be not what many modules uses by default (8081,8081 and okapi itself 9130). If we use 9230-9239, then at least it's same range as Okapi. Of course, it would be even better with random ports for Okapi and here, but a lot of work for little gain.